src/components/SelectPokemons/SelectPokemons.tsx

```diff
@@ -0,0 +1,47 @@
+// import { useState } from 'react';
+import { DISABLED_POKEMONS } from '../../App';
+import { PokemonsData } from '../../types/PokemonsData';
+import { PokemonCard } from '../PokemonCard/PokemonCard';
+import './SelectPokemons.css';
+
+type Props = {
+  pokemons: PokemonsData[];
+  selectedPokemons: string[];
+  setSelectedPokemons: (arg0: string[]) => void;
+};
+
+const SELECTED_POKEMON_SIZE = 2;
+
+export function SelectPokemons({ pokemons, selectedPokemons, setSelectedPokemons }: Props): JSX.Element {
+  const onSelectPokemons = (pokemonName: string) => {
+    if (DISABLED_POKEMONS.includes(pokemonName)) return;
+    if (selectedPokemons.length >= SELECTED_POKEMON_SIZE) {
+      selectedPokemons.shift() ?? '';
+
+      setSelectedPokemons([...selectedPokemons, pokemonName]);
+    } else setSelectedPokemons([...selectedPokemons, pokemonName]);
+  };
```

Comment:
Selection logic is hardcoded into the component that is not good.

How to create two different `SelectPocemons` components one that is able to select 3 pokemons and Pikachu. Another 2 pokemons but not Pikachu.